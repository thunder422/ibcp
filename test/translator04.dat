#################################################
#  Translator Test #4: Substrings (2010-05-19)  #
#################################################

LEFT$(A$,1) = B$
LEFT$(A$,1) = B$ + C$
MID$(A$,2) = B$
MID$(A$,2) = B$ + C$
MID$(A$,2,3) = B$
MID$(A$,2,3) = B$ + C$
RIGHT$(A$,2) = B$
RIGHT$(A$,2) = B$ + C$
LEFT$(A$,1) = RIGHT$(B$,1)
LEFT$(A$,1) = RIGHT$(B$+C$,1)

# begin of error tests
RIGHT$(A$,2) = B
RIGHT$(A$,2) = B%
RIGHT$(A,2) = B$
RIGHT$(A%,2) = B$
RIGHT$(A$+B$,2) = C$
RIGHT$(LEFT$(A$,1),2) = B$
RIGHT$(LEFT$(A$+B$,1),2) = C$

# list assignments (2010-05-22)
LEFT$(A$,1),RIGHT$(B$,1) = C$
LEFT$(A$,1),MID$(B$,2,1),RIGHT$(C$,1) = D$
LEFT$(A$,1),B$ = C$
LEFT$(A$,1),B$,C$ = D$
A$,LEFT$(B$,1) = C$
A$,LEFT$(B$,1),C$ = D$
A$,B$,C$ = D$
A$,B$,LEFT$(C$,1) = D$
LEFT$(A$,1),B$,RIGHT$(C$,1) = D$

# more related tests (2010-10-05)
A(int(B))=C
