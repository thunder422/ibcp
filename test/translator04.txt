
Input: LEFT$(A$,1) = B$
Output: A$<ref> 1% B$ AssignLeft 

Input: LEFT$(A$,1) = B$ + C$
Output: A$<ref> 1% B$ C$ +$ AssignLeft 

Input: MID$(A$,2) = B$
Output: A$<ref> 2% B$ AssignMid2 

Input: MID$(A$,2) = B$ + C$
Output: A$<ref> 2% B$ C$ +$ AssignMid2 

Input: MID$(A$,2,3) = B$
Output: A$<ref> 2% 3% B$ AssignMid3 

Input: MID$(A$,2,3) = B$ + C$
Output: A$<ref> 2% 3% B$ C$ +$ AssignMid3 

Input: RIGHT$(A$,2) = B$
Output: A$<ref> 2% B$ AssignRight 

Input: RIGHT$(A$,2) = B$ + C$
Output: A$<ref> 2% B$ C$ +$ AssignRight 

Input: LEFT$(A$,1) = RIGHT$(B$,1)
Output: A$<ref> 1% B$ 1% RIGHT$( AssignLeft 

Input: LEFT$(A$,1) = RIGHT$(B$+C$,1)
Output: A$<ref> 1% B$ C$ +$ 1% RIGHT$( AssignLeft 

Input: RIGHT$(A$,2) = B
                      ^-- expected string expression

Input: RIGHT$(A$,2) = B%
                      ^^-- expected string expression

Input: RIGHT$(A,2) = B$
              ^-- expected string variable

Input: RIGHT$(A%,2) = B$
              ^^-- expected string variable

Input: RIGHT$(A$+B$,2) = C$
                ^-- expected comma

Input: RIGHT$(LEFT$(A$,1),2) = B$
              ^^^^^^-- expected string variable

Input: RIGHT$(LEFT$(A$+B$,1),2) = C$
              ^^^^^^-- expected string variable

Input: LEFT$(A$,1),RIGHT$(B$,1) = C$
Output: A$<ref> 1% B$<ref> 1% C$ AssignKeepRight AssignLeft 

Input: LEFT$(A$,1),MID$(B$,2,1),RIGHT$(C$,1) = D$
Output: A$<ref> 1% B$<ref> 2% 1% C$<ref> 1% D$ AssignKeepRight AssignKeepMid3 AssignLeft 

Input: LEFT$(A$,1),B$ = C$
Output: A$<ref> 1% B$<ref> C$ AssignKeep$ AssignLeft 

Input: LEFT$(A$,1),B$,C$ = D$
Output: A$<ref> 1% B$<ref> C$<ref> D$ AssignKeep$ AssignKeep$ AssignLeft 

Input: A$,LEFT$(B$,1) = C$
Output: A$<ref> B$<ref> 1% C$ AssignKeepLeft Assign$ 

Input: A$,LEFT$(B$,1),C$ = D$
Output: A$<ref> B$<ref> 1% C$<ref> D$ AssignKeep$ AssignKeepLeft Assign$ 

Input: A$,B$,C$ = D$
Output: A$<ref> B$<ref> C$<ref> D$ AssignList$ 

Input: A$,B$,LEFT$(C$,1) = D$
Output: A$<ref> B$<ref> C$<ref> 1% D$ AssignKeepLeft AssignKeep$ Assign$ 

Input: LEFT$(A$,1),B$,RIGHT$(C$,1) = D$
Output: A$<ref> 1% B$<ref> C$<ref> 1% D$ AssignKeepRight AssignKeep$ AssignLeft 

Input: A(int(B))=C
Output: B INT( CvtInt A(<ref> C Assign 

