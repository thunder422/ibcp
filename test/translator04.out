
Input: LEFT$(A$,1) = B$
Output: LEFT$(A$, 1) = B$ 

Input: LEFT$(A$,1) = B$ + C$
Output: LEFT$(A$, 1) = B$ + C$ 

Input: MID$(A$,2) = B$
Output: MID$(A$, 2) = B$ 

Input: MID$(A$,2) = B$ + C$
Output: MID$(A$, 2) = B$ + C$ 

Input: MID$(A$,2,3) = B$
Output: MID$(A$, 2, 3) = B$ 

Input: MID$(A$,2,3) = B$ + C$
Output: MID$(A$, 2, 3) = B$ + C$ 

Input: RIGHT$(A$,2) = B$
Output: RIGHT$(A$, 2) = B$ 

Input: RIGHT$(A$,2) = B$ + C$
Output: RIGHT$(A$, 2) = B$ + C$ 

Input: LEFT$(A$,1) = RIGHT$(B$,1)
Output: LEFT$(A$, 1) = RIGHT$(B$, 1) 

Input: LEFT$(A$,1) = RIGHT$(B$+C$,1)
Output: LEFT$(A$, 1) = RIGHT$(B$ + C$, 1) 

Input: RIGHT$(A$,2) = B
                      ^-- expected string expression

Input: RIGHT$(A$,2) = B%
                      ^^-- expected string expression

Input: RIGHT$(A,2) = B$
              ^-- expected string variable

Input: RIGHT$(A%,2) = B$
              ^^-- expected string variable

Input: RIGHT$(A$+B$,2) = C$
                ^-- expected comma

Input: RIGHT$(LEFT$(A$,1),2) = B$
              ^^^^^^-- expected string variable

Input: RIGHT$(LEFT$(A$+B$,1),2) = C$
              ^^^^^^-- expected string variable

Input: LEFT$(A$,1),RIGHT$(B$,1) = C$
Output: LEFT$(A$, 1), RIGHT$(B$, 1) = C$ 

Input: LEFT$(A$,1),MID$(B$,2,1),RIGHT$(C$,1) = D$
Output: LEFT$(A$, 1), MID$(B$, 2, 1), RIGHT$(C$, 1) = D$ 

Input: LEFT$(A$,1),B$ = C$
Output: LEFT$(A$, 1), B$ = C$ 

Input: LEFT$(A$,1),B$,C$ = D$
Output: LEFT$(A$, 1), B$, C$ = D$ 

Input: A$,LEFT$(B$,1) = C$
Output: A$, LEFT$(B$, 1) = C$ 

Input: A$,LEFT$(B$,1),C$ = D$
Output: A$, LEFT$(B$, 1), C$ = D$ 

Input: A$,B$,C$ = D$
Output: A$, B$, C$ = D$ 

Input: A$,B$,LEFT$(C$,1) = D$
Output: A$, B$, LEFT$(C$, 1) = D$ 

Input: LEFT$(A$,1),B$,RIGHT$(C$,1) = D$
Output: LEFT$(A$, 1), B$, RIGHT$(C$, 1) = D$ 

Input: A(int(B))=C
Output: A(INT(B)) = C 

